src/type.h

```diff
@@ -116,6 +125,10 @@ class Type {
     return static_cast<Index>(enum_);
   }
 
+  bool IsTypeIdx() const {
+    return IsIndex() && (enum_ != Type::ExternRef || enum_ == Type::FuncRef);
```

Comment:
Isn't the second condition always true if the first one is? (because `IsIndex` implies the `enum_` is zero/positive and thus not a negative value like `ExternRef`)

Reply:
You are right, it was a typo, thanks for the catch